### PR TITLE
Devinabox

### PR DESCRIPTION
**Pull request by whenhellfreezes**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

This PR merges in the changes necessary for making it easy to run minikube on an AWS instance. This is to integrate with the DevInABox repo https://github.com/DecipherNow/devinabox which allows for users to make an EC2 instance with all dependencies setup automatically using packer.

<br/><br/>

2. What **changes to custom.yaml** are required?

None

<br/><br/>

3. Have you documented any additional setup steps or configurations options?

There is documentation in https://github.com/DecipherNow/devinabox

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Run make in https://github.com/DecipherNow/devinabox

<br/><br/>

### CODE REVIEW

**Comment by chrisbsmith** [comment]
@whenhellfreezes it doesn't look like this file is complete: https://github.com/DecipherNow/devinabox/blob/master/packerPlay.yml

Can you please review

**Comment by whenhellfreezes** [comment]
> @whenhellfreezes it doesn't look like this file is complete: https://github.com/DecipherNow/devinabox/blob/master/packerPlay.yml
> 
> Can you please review

The setup action can take an empty argument. Since it is the last command in the pre-tasks the playbook will continue into the roles after. It being the last line of the file however may give a wrong visual impression. Did you run it and see an error?

File: ci/scripts/minikube.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+cd $(dirname "${BASH_SOURCE[0]}")/../..
+
+#Determine if we are on AWS or not
+MINI='minikube'
+LC=$(curl -s -m 2 169.254.169.254/latest/meta-data | wc -l )
+if [ $LC -ge 4 ]; then
+    MINI='sudo /home/ubuntu/bin/minikube'
```
Comment (kaitmore) [inline ci/scripts/minikube.sh]: This assumes the location of the minikube binary, perhaps we could use the output of `which minikube` to make this a bit more flexible?

Comment (whenhellfreezes) [reply ci/scripts/minikube.sh]: Good idea.

**Comment by whenhellfreezes** [comment]
> LGTM.
> 
> Just a thought, it could be nice to get this to work on a vanilla Ubuntu AMI. The only things I had to do to run `make minikube` that way is:
> 
>     * put the minikube binary in `home/ubuntu/bin/minikube` and add that to PATH
> 
>     * add aliases for minikube, helm, and kubectl'
> 
>     * update the `minikube start` command to use the `--vm-driver=none` flag

For some reason I though vm-driver none needed to run minikube as sudo. I am going to try and simplify this a bit.

**Comment by whenhellfreezes** [comment]
> Awesome, looks great! Do we want to update the Deploy with Minikube/ec2 docs? This is what I had to run to get GM up on ec2:
> 
> ```
> sudo apt update
> sudo apt install make
> sudo apt install docker.io socat -y
> sudo snap install kubectl --classic
> curl -L https://git.io/get_helm.sh | bash -s -- --version v2.16.1
> wget https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
> chmod +x minikube-linux-amd64
> sudo mv minikube-linux-amd64 /usr/local/bin/minikube
> git clone --single-branch --branch devinabox https://github.com/DecipherNow/helm-charts --quiet
> cd helm-charts
> make fresh
> ```

Sure, have you tried out the flow of the actual devinabox repo? The flow will essentially be make -> ssh ubuntu@ec2ip -> cd helm-charts -> make fresh.

**Comment by kaitmore** [comment]
@whenhellfreezes Oh, no I haven't. Maybe we should just provide a link to the devinabox repo in the instructions then.

**Comment by whenhellfreezes** [comment]
> @whenhellfreezes Oh, no I haven't. Maybe we should just provide a link to the devinabox repo in the instructions then.

Added link.

**Pull request merged**